scipy/sparse/_coo.py

```diff
@@ -186,8 +186,10 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
                     if check_shape(shape) != self._shape:
                         raise ValueError('inconsistent shapes: %s != %s' %
                                          (shape, self._shape))
-
-                self.row, self.col = M.nonzero()
+                index_dtype = self._get_index_dtype(maxval=max(self._shape))
+                row, col = M.nonzero()
+                self.row = row.astype(index_dtype)
+                self.col = col.astype(index_dtype)
```

Comment:
Let's pass `copy=False` here, for both astype calls.

Reply:
Good point, fixed.